go/mysql/server.go

```diff
@@ -335,60 +338,39 @@ func (l *Listener) handle(conn net.Conn, connectionID uint32, acceptTime time.Ti
 			queryStart := time.Now()
 			query := c.parseComQuery(data)
 			c.recycleReadPacket()
-			fieldSent := false
-			// sendFinished is set if the response should just be an OK packet.
-			sendFinished := false
-			err := l.handler.ComQuery(c, query, func(qr *sqltypes.Result) error {
-				if sendFinished {
-					// Failsafe: Unreachable if server is well-behaved.
-					return io.EOF
-				}
-
-				if !fieldSent {
-					fieldSent = true
-
-					if len(qr.Fields) == 0 {
-						sendFinished = true
-						// We should not send any more packets after this.
-						return c.writeOKPacket(qr.RowsAffected, qr.InsertID, c.StatusFlags, 0)
-					}
-					if err := c.writeFields(qr); err != nil {
-						return err
-					}
-				}
-
-				return c.writeRows(qr)
-			})
-
-			// If no field was sent, we expect an error.
-			if !fieldSent {
-				// This is just a failsafe. Should never happen.
-				if err == nil || err == io.EOF {
-					err = NewSQLErrorFromError(errors.New("unexpected: query ended without no results and no error"))
-				}
+			stmts, err := sqlparser.SplitStatementToSlice(query)
+			if err != nil {
 				if werr := c.writeErrorPacketFromError(err); werr != nil {
 					// If we can't even write the error, we're done.
 					log.Errorf("Error writing query error to %s: %v", c, werr)
 					return
 				}
 				continue
 			}
-
-			if err != nil {
-				// We can't send an error in the middle of a stream.
-				// All we can do is abort the send, which will cause a 2013.
-				log.Errorf("Error in the middle of a stream to %s: %v", c, err)
-				return
+			if len(stmts) > 1 && c.Capabilities&CapabilityClientMultiStatements == 0 {
+				cerr := &SQLError{
+					Num:     ERParseError,
+					State:   ErrParse,
+					Message: fmt.Sprintf(multiStatementError, query),
```

Comment:
Including the non-normalized query in an error message is a problem for potential PII leakage, not to mention the fact that the query could be huge in the case of a big bulk insert.

IMO we should just return a static error string without including the query.